service/src/main/java/bio/terra/workspace/service/workspace/model/GcpCloudContext.java

```diff
@@ -1,61 +1,57 @@
 package bio.terra.workspace.service.workspace.model;
 
 import bio.terra.workspace.common.exception.InternalLogicException;
-import bio.terra.workspace.db.DbSerDes;
 import bio.terra.workspace.db.model.DbCloudContext;
 import bio.terra.workspace.generated.model.ApiGcpContext;
-import bio.terra.workspace.service.resource.model.WsmResourceState;
-import bio.terra.workspace.service.workspace.exceptions.InvalidSerializedVersionException;
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import java.util.Optional;
 import javax.annotation.Nullable;
 
 public class GcpCloudContext implements CloudContext {
-  private final String gcpProjectId;
-  private final String samPolicyOwner;
-  private final String samPolicyWriter;
-  private final String samPolicyReader;
-  private final String samPolicyApplication;
-  private final @Nullable CloudContextCommonFields commonFields;
+  private final @Nullable GcpCloudContextFields contextFields;
+  private final CloudContextCommonFields commonFields;
 
   @JsonCreator
   public GcpCloudContext(
-      @JsonProperty("gcpProjectId") String gcpProjectId,
-      @JsonProperty("samPolicyOwner") String samPolicyOwner,
-      @JsonProperty("samPolicyWriter") String samPolicyWriter,
-      @JsonProperty("samPolicyReader") String samPolicyReader,
-      @JsonProperty("samPolicyApplication") String samPolicyApplication,
-      @JsonProperty("commonFields") @Nullable CloudContextCommonFields commonFields) {
-    this.gcpProjectId = gcpProjectId;
-    this.samPolicyOwner = samPolicyOwner;
-    this.samPolicyWriter = samPolicyWriter;
-    this.samPolicyReader = samPolicyReader;
-    this.samPolicyApplication = samPolicyApplication;
+      @JsonProperty("contextFields") @Nullable GcpCloudContextFields contextFields,
+      @JsonProperty("commonFields") CloudContextCommonFields commonFields) {
```

Comment:
IIUC this will break in-progress "createGcpContext" flights. Those can sometimes run longer (10-15m if DRS is grumpy), so we should probably be aware of this when deploying to prod

Reply:
Good  point. These changes will break things in flight.
We should make a plan. I am thinking that this will go into preprod next Tuesday and then go into prod on Monday, June 5. Maybe it would make sense for me to do the preprod -> prod upgrade on Sunday evening (June 4).
wdyt?

Reply:
I'd either lean towards an off-hours release or just verifying there aren't any active flights in prod when we start the release